presto-cli/src/main/java/com/facebook/presto/cli/TsvPrinter.java

```diff
@@ -55,6 +55,13 @@ public void printRows(List<List<?>> rows, boolean complete)
     public void finish()
             throws IOException
     {
+        flush();
+    }
+
+    @Override
+    public void flush()
+            throws IOException
+    {
         printRows(ImmutableList.<List<?>>of(), true);
```

Comment:
There is definitely some hacky code in the CLI that could use some refactoring :)